### PR TITLE
Left Side Panel Not Updating After Next Lesson PassesBranch next lesson

### LINKED ISSUE #193: Bug: Left Side Panel Not Updating After Next Lesson Passes
### Description
When the current time is close to the next scheduled lesson, and that time passes, the left side panel does not automatically update to show the following lesson. This results in outdated lesson information being displayed.

### Steps to Reproduce
1. Set the current time close to the next lesson time.
2. Wait for the lesson time to pass.
3. Observe that the left side panel does not update to reflect the next lesson.

### Expected Behavior
The left side panel should automatically update as soon as the scheduled lesson time has passed to show the upcoming lesson.

### Current Workaround
Currently, the user must manually refresh or navigate away and back for the left side panel to update.

### Proposed Solution
- Implement a timer to check every minute if the current lesson time has passed.
- If the time has passed, trigger an update to the left side panel to display the next lesson.

### PR DESCRIPTION
**Pull request by weijianwong**
Resolve #193
### What is this pull request for?
This pull request updates PersonCard to refresh the next lesson display every minute. It also modifies PersonListPanel to stop the existing timeline in a PersonCard when a Person is edited, preventing multiple overlapping timelines which could lead to redundant logging and performance issues.

### What will this pull request do?
- Added a method in PersonCard to refresh the next lesson every minute.
- Updated PersonListPanel to stop the refresh timeline in PersonCard if the Person entry is edited.

### CODE REVIEW

File: src/main/java/tuteez/ui/PersonListPanel.java [hunk 1]
```diff
@@ -27,23 +34,86 @@ public PersonListPanel(ObservableList<Person> personList) {
...
         personListView.setCellFactory(listView -> new PersonListViewCell());
+        startCentralRefreshTimeline();
     }
 
+    /**
+     * Starts the central timeline that refreshes all visible PersonCards every minute.
+     */
+    private void startCentralRefreshTimeline() {
+        if (centralRefreshTimeline != null) {
+            centralRefreshTimeline.stop();
```
Comment (curtischang2510) [inline src/main/java/tuteez/ui/PersonListPanel.java:45]: Can i confirm that if this code reaches this point is there supposed to be some bug? because somehow this method is called more than once per panel. If that was the case maybe you write an assert statement here or log it just in case. 

Additionally perhaps it is better to not `stop()` but `return`? Do you expect behaviour of the earlier `centralRefreshTime` to be different till where you need to reset everything?

**Review by curtischang2510 — COMMENTED** [review]
I've tried your code alr, seems like it works perhaps we should test it along boundary cases such as midnight, maybe we can try that today
